keps/sig-api-machinery/2340-Consistent-reads-from-cache/README.md

```diff
@@ -207,6 +216,13 @@ Since falling back to etcd won't work, we should fail the requests and rely on
 rate limiting to prevent cascading failure.  I.e. `Retry-After` HTTP header (for
 well behaved clients) and [Priority and Fairness](https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md).
 
+For such situations we will provide a dedicated metric to detect a problem,
+and a per-request override for debugging. If there is a ongoing problem,
+user can reconfigure kube-apiserver to always pass request to etcd by 
+disabling the feature flag. Metric `apiserver_watch_cache_read_wait`
+will measure wait time experienced by reads for watch cache to become fresh.
+Per request override will be done via a HTTP request header `k8s.io/apiserver/skip-watch-cache`.
```

Comment:
Using HTTP headers is not a typical thing to do - the typical thing to do is to extend `ListOptions`

I think it should be ListOptions field, but I will defer the actual response to @deads2k from API approval perspective.

Reply:
Agree, this is not typical. However, we are about talking about change that is transparent from content perspective. Want to confirm that we need to have a full API change just for debugging.

Reply:
@deads2k - I would like to get a feedback about it from you

Reply:
Any HTTP header option needs to be described in `ListOptions` (or the appropriate options struct).  This ensures that we know our API surface, it is consistent, and our storage layers properly handle the input.

Once included in our API surface area, this option would persist forever.

Reply:
As this is a override for the debugging that doesn't change the semantic of the request, it might be reasonable to not include it in `ListOptions`.